contracts/Gauge.sol

```diff
@@ -16,25 +18,23 @@ import "./interfaces/IVotingEscrow.sol";
     @dev this contract is used behind multiple delegate proxies.
  */
 
-contract Gauge is BaseGauge, IGauge {
+contract Gauge is BaseGauge, ERC20Upgradeable, IGauge {
```

Comment:
I would create the `IERC4626` interface and use it here to make the `IERC4626 compliant` explicit for external devs/users.

Reply:
IGauge is inheriting from IERC4626, isn't it enough?
```
pragma solidity 0.8.14;
import "./IBaseGauge.sol";
import "./IERC4626.sol";

interface IGauge is IBaseGauge, IERC4626 {
```

Reply:
> IGauge is inheriting from IERC4626, isn't it enough?
> 
> ```
> pragma solidity 0.8.14;
> import "./IBaseGauge.sol";
> import "./IERC4626.sol";
> 
> interface IGauge is IBaseGauge, IERC4626 {
> ```

Good point, sorry, I missed that one.